### ISSUE TITLE
`jsx-no-constructed-context-values`: false positive

### ISSUE THREAD

**Issue by thany** [yannickcr/eslint-plugin-react#3188]
Consider this code:

```js
export const AzureApplicationGatewayContext = createContext({});

const AzureApplicationGatewayProvider = ({ children, sitecoreContext }) => {
  const {
    azureApplicationGateway: {
      isEnabled = false,
      paths = [],
    } = {},
  } = sitecoreContext;

  const isInternalLink = useMemo(() => {
    if (!isEnabled) { return () => true; }

    return createInternalLinkMatcher(paths);
  }, [isEnabled, paths]);

  return (
    <AzureApplicationGatewayContext.Provider value={{ isInternalLink }}>
      {/*                                           ^^^^^^^^^^^^^^^^^^ */}
      {children}
    </AzureApplicationGatewayContext.Provider>
  );
};
```

Don't worry about Azure. We're coping with it 😅

`isInternalLink` here, is being marked with:

> The object passed as the value prop to the Context provider (at line 50) changes every render. To fix this consider wrapping it in a useMemo hook.
> eslint(react/jsx-no-constructed-context-values)

I *am* wrapping it in a useMemo hook, aren't I? 🤨 Then why is it saying I should do exactly the thing I've already done? 🤔

For now, I think I'll just disable this rule. I'm in the middle of fixing other linting errors (which are true positives 😉) so maybe there's a workaround, which I'll be happy to follow if it's reasonable.

**Comment by golopot** [comment]
This is a valid warning to a real problem. The variable `isInternalLink` is memo-ed, but `{ isInternalLink }` is a new object in every render. See https://reactjs.org/docs/context.html#caveats .

**Comment by ljharb** [comment]
Exactly right, this is indeed a true positive.

**Comment by thany** [comment]
😑 I've always learned that React is okay with passing multiple values in ad-hoc objects. Why not in this case? Come on, React.

Anyway, that's not on you. Actually, thanks for taking the time to look into this 👍🏻

Maybe something can be done about the message that's emitted, to make it clear that it's about the entire object. When passing an object with a single value, you can see it's easy to create confusion between the value and the object wrapping it.

**Comment by ljharb** [comment]
Since the difference between those two is a basic JS thing, what prose would you suggest that would have made that more clear?

**Comment by thany** [comment]
This might do it for me.

> The whole object, not just the one value, passed as the value prop to the Context provider (at line 50) changes every render. To fix this consider wrapping it in a useMemo hook. See https://reactjs.org/docs/context.html#caveats for further reading.

Actually, I think the reference to the documentation might be the most important bit. When you pointed to it, it took probably less than 5 seconds to realise my mistake 😀

**Comment by ljharb** [comment]
A PR adding that link would be most welcome.
